cargo-shuttle/src/lib.rs

```diff
@@ -850,31 +850,23 @@ impl Shuttle {
 
     async fn logs_beta(&self, args: LogsArgs) -> Result<CommandOutcome> {
         let client = self.client.as_ref().unwrap();
-        let mut stream = client
-            // TODO: use something else than a fake Uuid
-            .get_logs_ws(self.ctx.project_name(), &Uuid::new_v4(), LogsRange::All)
-            .await
-            .map_err(|err| {
-                suggestions::logs::get_logs_failure(err, "Connecting to the logs stream failed")
-            })?;
-
-        while let Some(Ok(msg)) = stream.next().await {
-            if let tokio_tungstenite::tungstenite::Message::Text(line) = msg {
-                match serde_json::from_str::<shuttle_common::LogItemBeta>(&line) {
-                    Ok(log) => {
-                        if args.raw {
-                            println!("{}", log.line);
-                        } else {
-                            println!("{log}");
-                        }
-                    }
-                    Err(err) => {
-                        // TODO better handle logs, by returning a different type than the log line
-                        // if an error happened.
-                        bail!(err);
-                    }
-                }
-            }
+        let proj_name = self.ctx.project_name();
+        let logs = if args.all_deployments {
+            client.get_project_logs_beta(proj_name).await?.logs
+        } else {
+            // TODO:
+            // let depl = client.get_current_deployment_beta(proj_name).await?;
+            let depls = client.get_deployments_beta(proj_name).await?;
+            let depl = depls
+                .first()
+                .expect("at least one deployment in this project");
```

Comment:
suggestion(non-blocking): we might want to return a proper error to the user instead of panicking. Can be done in an other PR.